lib/big_query/sync/geonames_task.ex

```diff
@@ -0,0 +1,125 @@
+defmodule Mix.Tasks.Bigquery.Sync.Geonames do
+  use Mix.Task
+  require Logger
+  import Castle.Redis.Lock
+  alias NimbleCSV.RFC4180, as: CSV
+
+  @shortdoc "Sync maxmind geonames to BigQuery"
+  @lock "lock.bigquery.sync.geonames"
+  @lock_ttl 120
+  @success_ttl 30
+  @url "https://download.maxmind.com/app/geoip_download?edition_id=GeoLite2-City-CSV&suffix=zip&license_key="
+  @filename "GeoLite2-City-Locations-en.csv"
+
+  def run(args) do
+    {:ok, _started} = Application.ensure_all_started(:castle)
+
+    lock = Enum.member?(args, "--lock") || Enum.member?(args, "-l")
+
+    if lock do
+      lock(@lock, @lock_ttl, @success_ttl, do: sync_geonames())
+    else
+      sync_geonames()
+    end
+  end
+
+  defp sync_geonames do
+    case get_maxmind_geonames() do
+      {:ok, folder, count, data} ->
+        dest = "#{Env.get(:bq_dataset)}.geonames_tmp"
+        Logger.info("BigQuery.Sync.Geonames reload: #{dest} with #{count} rows from #{folder}")
+
+        # create tmp table and load data into it
+        BigQuery.Base.Query.run(create_table_sql("geonames_tmp"))
```

Comment:
Slight hiccup here - the Castle users need `bigquery.tables.create` for this to succeed.  Which would be fine - but then the next line fails because the user doesn't have `bigquery.tables.updateData` on that brand new table, or on the dataset.

I'd rather not give Castle write access to _everything_ in the dataset (mostly worried about the irreplaceable `dt_downloads` / `dt_impressions` tables).

So - I guess I'll have to make this tmp table permanent, and we'll all just have to look at it in the list of tables UI?